src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs

```diff
@@ -25,18 +25,18 @@
 //
 //------------------------------------------------------------------------------
 
-using Microsoft.IdentityModel.Logging;
-using Newtonsoft.Json;
 using System;
 using System.Collections.Generic;
+using Microsoft.IdentityModel.Logging;
+using Newtonsoft.Json;
 
-namespace System.IdentityModel.Tokens.Jwt
+namespace Microsoft.IdentityModel.Tokens
 {
     /// <summary>
     /// Represents a Json Web Key as defined in http://tools.ietf.org/html/rfc7517.
     /// </summary>
     [JsonObject]
-    public class JsonWebKey
+    public class JsonWebKey : AsymmetricSecurityKey
```

Comment:
Yes, they can be represented as JWK. But it is not very common to go to a metadata endpoint to retrieve a symmetric key. This usually has value in case of asymmetric keys. I had to split out the JsonWebKey into asymmetric and symmteric parts, because keeping them together was messing the OM when trying to create signature provider from JsonWebKey. While doing so, I didn't see enough value in supporting symmetric keys as JWK, and hence just kept the asymmetric parts in it. Do you think retrieving symmetric JWK from metadata endpoint is a common scenario? In that case, I might have to consider renaming those to AsymmetricJsonWebKey and SymmetricJsonWebKey.